src/lib/bedtools.ml

```diff
@@ -38,10 +57,47 @@ let bamtofastq ~(run_with:Machine.t) ~sample_type ~processors ~output_prefix inp
   ] |> fun list ->
     begin match r2opt with
     | None -> list
-    | Some r2 -> 
+    | Some r2 ->
       on_failure_activate (Remove.file ~run_with r2) :: list
     end
   in
   workflow_node
     (fastq_reads ~host:(Machine.as_host run_with) r1 r2opt)
     ~edges ~name ~make
+
+
+(** Used to determine if features in multiple sets intersect with one another.
+
+Feature sets include BED, VCF, GFF, and BAM files.
+
+- [a]: The primary set file (workflow_node with #path).
+- [bs]: List of set file workflow_nodes to intersect with.
+- [result]: Path to the resulting set.
+
+**)
+let intersect
+    ~(run_with:Machine.t)
+    ?(configuration=Configuration.Intersect.default)
+    ~a ~bs ~result
```

Comment:
Why not call them `~primary` and `~interset_with`?

The rest of the library seeems to use the word `output` instead of `result` (which reminds of `Result.t`).

Also when there are optional arguments we need one non-unnamed, please remove the `~` for result or add a terminating `()`.

Thanks for the proper documentation !